### PR TITLE
GSS Closed Loop BZ-2004130: Validate old noobaa-db service in cli and logs

### PR DESCRIPTION
**Pull request by parikshithb**
Signed-off-by: Parikshith Byregowda <pbyregow@redhat.com>

### CODE REVIEW

**Review by Akarsha-rai — DISMISSED** [review]

**Review by yosibsh — DISMISSED** [review]
LGTM

File: tests/manage/mcg/test_verify_noobaa_status.py [hunk 1]
```diff
@@ -20,4 +24,16 @@ def test_verify_noobaa_status_cli(mcg_obj_session):
     for line in status.split("\n"):
         if "Not Found" in line:
             assert "Optional" in line, f"Error in noobaa status output- {line}"
+        if float(config.ENV_DATA["ocs_version"]) > 4.7 and "Noobaa-db" in line:
```
Comment (OdedViner) [inline tests/manage/mcg/test_verify_noobaa_status.py:27]: you can use semantic method
``` 
ocs_version_semantic = version.get_semantic_ocs_version_from_config()
if ocs_version_semantic >= version.VERSION_4_7
```

File: tests/manage/mcg/test_verify_noobaa_status.py [hunk 2]
```diff
@@ -20,4 +24,16 @@ def test_verify_noobaa_status_cli(mcg_obj_session):
...
+            assert (
+                "Old noobaa-db service is logged" in line
+            ), f"Error in MCG Cli status output- {line}"
     log.info("Verified: noobaa status does not contain any error.")
+
+    # verify noobaa db logs for #bz2004130
+    if float(config.ENV_DATA["ocs_version"]) > 4.7:
```
Comment (OdedViner) [inline tests/manage/mcg/test_verify_noobaa_status.py:34]: you can use semantic method
```
ocs_version_semantic = version.get_semantic_ocs_version_from_config()
if ocs_version_semantic >= version.VERSION_4_7
```

File: tests/manage/mcg/test_verify_noobaa_status.py (hunk 1 above)
Comment (MeridianExplorer) [reply tests/manage/mcg/test_verify_noobaa_status.py:27]: Adding on top of Oded's suggestion - 
```suggestion
        if float(config.ENV_DATA["ocs_version"]) > 4.7 and "noobaa-db" in line.lower():
```

Comment (parikshithb) [reply tests/manage/mcg/test_verify_noobaa_status.py]: Thanks both. Ack.

File: tests/manage/mcg/test_verify_noobaa_status.py (hunk 2 above)
Comment (parikshithb) [reply tests/manage/mcg/test_verify_noobaa_status.py]: Ack

**Review by Shrivaibavi — APPROVED** [review]
LGTM

File: tests/manage/mcg/test_verify_noobaa_status.py [hunk 3]
```diff
@@ -1,23 +1,41 @@
...
     Verify noobaa status output is clean without any errors using the noobaa cli
     """
+    ocs_version_semantic = version.get_semantic_ocs_version_from_config()
+
     # Get noobaa status
     status = mcg_obj_session.exec_mcg_cmd("status").stderr
     for line in status.split("\n"):
         if "Not Found" in line:
             assert "Optional" in line, f"Error in noobaa status output- {line}"
+        if ocs_version_semantic >= version.VERSION_4_7 and "noobaa-db" in line:
```
Comment (PrasadDesala) [inline tests/manage/mcg/test_verify_noobaa_status.py:29]: BZ 2004130 looks to be fixed in 4.8, should it be > 4.7 ?

File: tests/manage/mcg/test_verify_noobaa_status.py [hunk 4]
```diff
@@ -1,23 +1,41 @@
...
+            assert (
+                "Old noobaa-db service is logged" in line
+            ), f"Error in MCG Cli status output- {line}"
     log.info("Verified: noobaa status does not contain any error.")
+
+    # verify noobaa db logs for #bz2004130
+    if ocs_version_semantic >= version.VERSION_4_7:
```
Comment (PrasadDesala) [inline tests/manage/mcg/test_verify_noobaa_status.py:36]: same as above comment

File: tests/manage/mcg/test_verify_noobaa_status.py (hunk 3 above)
Comment (parikshithb) [reply tests/manage/mcg/test_verify_noobaa_status.py]: True, ack

File: tests/manage/mcg/test_verify_noobaa_status.py (hunk 4 above)
Comment (parikshithb) [reply tests/manage/mcg/test_verify_noobaa_status.py]: ack

**Review by PrasadDesala — APPROVED** [review]

**Review by Akarsha-rai — APPROVED** [review]

**Review by Shrivaibavi — APPROVED** [review]
LGTM

**Comment by PrasadDesala** [comment]
@parikshithb Your current branch has some conflicts, please rebase

**Review by Akarsha-rai — APPROVED** [review]

**Comment by parikshithb** [comment]
Had a discussion with Ben, he suggested to have this flow as a separate test since it checks logs as well whereas the original test  is better suited for cli status only.
I have made the changes.

File: tests/manage/mcg/test_verify_noobaa_status.py [hunk 5]
```diff
@@ -21,3 +26,31 @@ def test_verify_noobaa_status_cli(mcg_obj_session):
...
+def test_verify_noobaa_db_service(mcg_obj_session):
+    """
+    Validates whether MCG cli and noobaa db logs does not check 'noobaa-db'
+    """
+    # Get noobaa status
+    status = mcg_obj_session.exec_mcg_cmd("status").stderr
+    for line in status.split("\n"):
+        assert (
+            'Service "noobaa-db"' not in status
+        ), f"Error in MCG Cli status output- {line}"
```
Comment (MeridianExplorer) [inline tests/manage/mcg/test_verify_noobaa_status.py:46]:
```suggestion
    status = mcg_obj_session.exec_mcg_cmd("status").stderr
    assert (
        'Service "noobaa-db"' not in status
    ), f"Error in MCG Cli status output- {line}"
```

Comment (parikshithb) [reply tests/manage/mcg/test_verify_noobaa_status.py]: ack

**Review by MeridianExplorer — APPROVED** [review]

**Review by PrasadDesala — APPROVED** [review]

**Pull request merged**